### ISSUE TITLE
Linkerd2 request timeout do not use application timeout values

### ISSUE THREAD

**Issue by serkangunes** [linkerd/linkerd2#1975]
### What is the issue?
linkerd2 proxy adds a 10 second request timeout between two services.

### How can it be reproduced?

- Create two GRPC services one server, one client
- Add artificial delay that is greater than 10s two server handler
- Add data-plane two both of them
- Send a request from the client making sure request timeout is set to 30s.

### Expected Result
The request should time-out around 30 seconds.
Basically there shouldn't be any behavioural change between with linkerd2 and without linkerd2 configurations. Application should decide what is the timeout value.

### Actual Results
The request times out around 10 seconds.

### Possible work around
Use  --proxy-bind-timeout=30s parameter during the data plane injection.

**Comment by klingerf** [comment]
@serkangunes Thanks for reporting this, and for providing the workaround! We're in the process of removing the bind timeout as part of #2017, at which point this 10 second limitation will be removed.

**Comment by serkangunes** [comment]
@klingerf No problem. Thanks for your swift action :)

**Comment by wmorgan** [comment]
@klingerf since #2017 has been merged, can we close this out?

**Comment by klingerf** [comment]
Yep, this has been fixed. Will close.
